services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java

```diff
@@ -730,4 +659,51 @@ private void showOutcomeDialog(SyncStatus status, SyncOverallResult result) {
     }
   }
 
+  private void updatePropsOnSyncComplete(SyncStatus status){
+    Map<String, String> properties=new HashMap<>();
+    switch (status){
+      case SERVER_IS_NOT_ODK_SERVER:{
+        properties.put(CommonToolProperties.KEY_IS_SERVER_VERIFIED,Boolean.toString(false));
+        break;
+      }
+
+      case AUTHENTICATION_ERROR:{
+        properties.put(CommonToolProperties.KEY_IS_SERVER_VERIFIED, Boolean.toString(true));
+
+        if(syncViewModel.getUserState()==UserState.ANONYMOUS){
+          properties.put(CommonToolProperties.KEY_IS_ANONYMOUS_SIGN_IN_USED, Boolean.toString(true));
+          properties.put(CommonToolProperties.KEY_IS_ANONYMOUS_ALLOWED, Boolean.toString(false));
+        }
+        else {
+          properties.put(CommonToolProperties.KEY_IS_USER_AUTHENTICATED, Boolean.toString(false));
+        }
+        break;
+      }
+
+      case SYNC_COMPLETE:
+      case SYNC_COMPLETE_PENDING_ATTACHMENTS: {
+        properties.put(CommonToolProperties.KEY_IS_SERVER_VERIFIED, Boolean.toString(true));
+
+        if(syncViewModel.getUserState()==UserState.ANONYMOUS){
+          properties.put(CommonToolProperties.KEY_IS_ANONYMOUS_SIGN_IN_USED, Boolean.toString(true));
+          properties.put(CommonToolProperties.KEY_IS_ANONYMOUS_ALLOWED, Boolean.toString(true));
+        }
+        else {
+          properties.put(CommonToolProperties.KEY_IS_USER_AUTHENTICATED, Boolean.toString(true));
+        }
+
+        String timestamp = String.valueOf(System.currentTimeMillis());
+        properties.put(CommonToolProperties.KEY_LAST_SYNC_INFO,timestamp);
+
+        break;
+      }
+    }
+    updatePropertiesSingleton(properties);
+  }
+
+  private void updatePropertiesSingleton(Map<String, String> properties){
+    PropertiesSingleton props=((AbsSyncBaseActivity)requireActivity()).getProps();
+    props.setProperties(properties);
+    ((AbsSyncBaseActivity)requireActivity()).updateViewModelWithProps();
+  }
```

Comment:
It is even present in `LoginFragment`, can't we move this to abstract parent class and use it across all the fragment wherever required.